signer.go

```diff
@@ -0,0 +1,104 @@
+package iopipe
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+)
+
+// SignerRequest is a signer request
+type SignerRequest struct {
+	ARN       string `json:"arn"`
+	RequestID string `json:"requestId"`
+	Timestamp int    `json:"timestamp"`
+	Extension string `json:"extension"`
+}
+
+// SignerResponse is a signer response
+type SignerResponse struct {
+	JWTAccess     string `json:"jwtAccess"`
+	SignedRequest string `json:"signedRequest"`
+	URL           string `json:"url"`
+}
+
+// GetSignerURL returns the URL for the signer in the specified region
+func GetSignerURL(region string) string {
+	supportedRegions := map[string]struct{}{
+		"ap-northeast-1": struct{}{},
+		"ap-southeast-2": struct{}{},
+		"eu-west-1":      struct{}{},
+		"us-east-1":      struct{}{},
+		"us-east-2":      struct{}{},
+		"us-west-1":      struct{}{},
+		"us-west-2":      struct{}{},
+	}
+
+	if region == "mock" {
+		return os.Getenv("MOCK_SERVER")
+	}
+
+	if _, exists := supportedRegions[region]; exists {
+		return fmt.Sprintf("https://signer.%s.iopipe.com/", region)
+	}
+
+	return "https://signer.us-east-1.iopipe.com/"
```

Comment:
Should this default to east-1 or west-2? (I think we default to west-2 in newer services?)